app/types/app_version_type.rb

```diff
@@ -0,0 +1,15 @@
+class AppVersionType < ActiveRecord::Type::Value
+  def cast(value)
+    AppVersion.new(value) unless value.nil?
+  rescue StandardError
+    nil
+  end
+
+  def serialize(value)
+    value.to_pg_array if value
+  end
+
+  def deserialize(value)
+    AppVersion.from_pg_array(value) if value
```

Comment:
Again, the customer-facing casting from string is different than the database-facing casting-to-PG-array. It all works nicely, though, and we can just use DB comparison operators on pairs of arrays-representing-version-numbers.